test/testVegaWrapper.js

```diff
@@ -417,3 +418,351 @@ describe('vegaWrapper', function() {
+
+describe('vegaWrapper2', function() {
+    function expectError(testFunc, msg, errFuncNames) {
+        var error, result;
+        msg = JSON.stringify(msg);
+        try {
+            result = testFunc();
+        } catch (err) {
+            error = err;
+        }
+
+        if (!error) {
+            assert(false, util.format('%j was expected to cause an error in functions %j, but returned %j',
+                msg, errFuncNames, result));
+        }
+
+        if (error.stack.split('\n').map(function (v) {
+                return v.trim().split(' ');
+            }).filter(function (v) {
+                return v[0] === 'at';
+            })[0][1] in errFuncNames
+        ) {
+            // If first stack line (except the possibly multiline message) is not expected function, throw
+            error.message = '"' + msg + '" caused an error:\n' + error.message;
+            throw error;
+        }
+    }
+
+    var domains = {
+        http: ['nonsec.org'],
+        https: ['sec.org'],
+        wikiapi: ['wikiapi.nonsec.org', 'wikiapi.sec.org'],
+        wikirest: ['wikirest.nonsec.org', 'wikirest.sec.org'],
+        wikiraw: ['wikiraw.nonsec.org', 'wikiraw.sec.org'],
+        wikidatasparql: ['wikidatasparql.nonsec.org', 'wikidatasparql.sec.org'],
+        geoshape: ['maps.nonsec.org', 'maps.sec.org']
+    };
+    var domainMap = {
+        'nonsec': 'nonsec.org',
+        'sec': 'sec.org'
+    };
+
+    var wrapper = new VegaWrapper2({
+            loader: {},
+            domains: domains,
+            domainMap: domainMap,
+            logger: function (msg) { throw new Error(msg); },
+            formatUrl: urllib.format,
+            languageCode: 'en'
+        });
+
+    describe('sanitize', function () {
+        var pass = function (url, expected, addCorsOrigin) {
+                var opt = {domain: 'domain.sec.org'};
+                const result = wrapper.objToUrl(url, opt);
+                assert.equal(result, expected, JSON.stringify(url));
+                assert.equal(opt.addCorsOrigin, addCorsOrigin, 'addCorsOrigin');
+            },
+            passWithCors = function (url, expected) {
+                pass(url, expected, true);
+            },
+            fail = function (url) {
+                expectError(function () {
+                    return wrapper.objToUrl(url, {domain: 'domain.sec.org'});
+                }, url, ['VegaWrapper2.objToUrl', 'VegaWrapper2._overrideHostAndProtocol']);
+            };
+
+        it('error type', function () {
+            fail({});
+            fail({ path: 'blah' });
+            fail({ type: 'blah', title: 'MyPage' });
+            fail({ type: 'nope', host: 'sec.org' });
+            fail({ type: 'nope', host: 'sec' });
+            fail({ type: 'https', host: 'sec.org' });
+            fail({ type: 'https', host: 'sec' });
+        });
+
+        it('wikiapi', function () {
+            // wikiapi allows sub-domains
+            fail({ type: 'wikiapi', wiki: 'sec.org', });
```

Comment:
add fail test for params being non-object, or if params contains a field that is an object itself, e.g.
```
fail({type:'wikiapi', params: "bad"});
fail({type:'wikiapi', params: null});
fail({type:'wikiapi', params: {a: null}});
fail({type:'wikiapi', params: {a: {}}});
fail({type:'wikiapi', params: {a: {b: 2}}});
fail({type:'wikiapi', params: []});
fail({type:'wikiapi', params: {a: []}});